Sources/Sentry/SentrySystemEventBreadcrumbs.m

```diff
@@ -41,11 +45,30 @@ - (void)start
 - (void)stop
 {
 #if TARGET_OS_IOS
-    NSNotificationCenter *defaultCenter = [NSNotificationCenter defaultCenter];
-    [defaultCenter removeObserver:self];
+    // Remove the observers with the most specific detail possible, see
+    // https://developer.apple.com/documentation/foundation/nsnotificationcenter/1413994-removeobserver
+    [self.notificationCenterWrapper removeObserver:self name:UIKeyboardDidShowNotification];
```

Comment:
If we're removing observers via the wrapper object, shouldn't we also be adding observers using it instead of the defaultCenter used on lines 98, 158, 189, 215 and 234?

Reply:
Sure thing